ingest/utils.go

```diff
@@ -0,0 +1,55 @@
+/*************************************************************************
+ * Copyright 2022 Gravwell, Inc. All rights reserved.
+ * Contact: <legal@gravwell.io>
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD 2-clause license. See the LICENSE file for details.
+ **************************************************************************/
+
+package ingest
+
+import (
+	crand "crypto/rand"
+	"encoding/binary"
+	"math/rand"
+	"sync"
+)
+
+// The implementation of this is actually in the Go stdlib, it's just not exported
+// See math/rand/rand.go in the Go source tree.
+type LockedSource struct {
+	lk  sync.Mutex
+	src rand.Source
+}
+
+func NewLockedSource(src rand.Source) rand.Source {
+	return &LockedSource{src: src}
+}
+
+func (r *LockedSource) Int63() (n int64) {
+	r.lk.Lock()
+	n = r.src.Int63()
+	r.lk.Unlock()
+	return
+
+}
+
+func (r *LockedSource) Seed(seed int64) {
+	r.lk.Lock()
+	r.src.Seed(seed)
+	r.lk.Unlock()
+}
+
+func NewRNG() *rand.Rand {
+	return rand.New(NewLockedSource(rand.NewSource(SecureSeed())))
+}
+
+func NewInsecureRNG() *rand.Rand {
+	return rand.New(NewLockedSource(rand.NewSource(rand.Int63())))
+}
+
+func SecureSeed() int64 {
+	bts := make([]byte, 8)
+	crand.Read(bts)
```

Comment:
we should check this error and do something about it.  Failing a cryptographic reseed means a seed of zero which means a predictable PRNG.  Basically on a secure seed if we get an error keep hitting the secure random source until we get something back?  A bad cryptographic source is so bad i honestly think we panic here....

Lets do this:
1. Try to get a secure seed up to 3 times, if it fails all 3 times, go ahead and panic